### PR TITLE
deploy KubeSphere on AKS version 1.0

### PR DESCRIPTION
**Pull request by ruiyaoOps**
Signed-off-by: yaorui <yaorui@yunify.com>

# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* deployment kubesphere on aks version 1.0
*
*

### CODE REVIEW

File: aks-kubesphere-linux/azuredeploy.parameter.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "virtualNetworkName": {
+            "value": "KubeSphereRG-vnet"
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.parameter.json:6]: remove the literal values from here and use defaultValues in the template

File: aks-kubesphere-linux/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+            "type": "string",
+            "allowedValues": [
+                "1.17.13",
+                "1.17.16",
+                "1.18.10",
+                "1.18.14",
+                "1.19.6",
+                "1.19.7"
+            ],
+            "defaultValue": "1.18.14",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:117]: use the latest as the defaultValue - these deprecate pretty quickly

File: aks-kubesphere-linux/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+                "Premium_LRS"
+            ],
+            "defaultValue": "StandardSSD_LRS",
+            "metadata": {
+                "description": "Client os Disk type"
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "aksuser",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:190]: remove defaultValue

File: aks-kubesphere-linux/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+            "metadata": {
+                "description": "Client node admin user name."
+            }
+        },
+        "adminPassword": {
+            "type": "string",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:196]: use secureString for secrets

File: aks-kubesphere-linux/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+            "defaultValue": "ABCabc321",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:197]: remove defaultValue

File: aks-kubesphere-linux/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+            }
+        },
+        "SPName": {
+            "type": "string",
+            "metadata": {
+                "description": "Service Principal User Name."
+            }
+        },
+        "SPPassword": {
+            "type": "string",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:220]: use secureString for secrets

File: aks-kubesphere-linux/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+        },
+        "SubscriptionID": {
+            "type": "string",
+            "metadata": {
+                "description": "SubscriptionID."
+            }
+        },
+        "_artifactsLocation": {
+            "type": "string",
+            "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/aks-kubesphere-linux/",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:239]: the latest information I have suggests that https://raw.githubusercontent.com was not accessible from China - has this changed?

File: aks-kubesphere-linux/README.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
+# Install KubeSphere on Azure Kubernetes Cluster
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/README.md:1]: see pipeline results for changes that need to be made here

File: aks-kubesphere-linux/README.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+## Requirements
+1. Install Azure CLI
+[Install Azure CLI on Windows](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?tabs=azure-cli)
+[Install Azure CLI on macOS](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-macos)
+
+2. Sign in
+[Sign in with Azure CLI](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli)
+
+3. Create a service principal
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/README.md:16]: can use use MSI for this sample?

File: aks-kubesphere-linux/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+        "networkPolicy": "azure",
+        "ddosProtectionPlanEnabled": false,
+        "networkSecurityGroupID": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups/', parameters('NetworkSecurityGroupName'))]",
+        "virtualNetworkId": "[resourceId(resourceGroup().name,'Microsoft.Network/VirtualNetworks',parameters('virtualNetworkName'))]",
+        "AKSsubnetRefID": "[resourceId('Microsoft.Network/virtualNetworks/subnets',parameters('virtualNetworkName'),parameters('AKSSubnetName'))]",
+        "ClientsubnetRefID": "[resourceId('Microsoft.Network/virtualNetworks/subnets',parameters('virtualNetworkName'),parameters('ClientSubnetName'))]",
+        "ClientPublicIpAddressName": "ClientNode-PIP",
+        "ClientPublicIpAddressType": "Dynamic",
+        "ClientPublicIpAddressSku": "Basic",
+        "location": "[resourceGroup().location]"
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:269]: this must be a parameter

File: aks-kubesphere-linux/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,506 @@
...
+			"dependsOn": [
+				"[resourceId('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkName'))]"
+			],
+			"properties": {
+				"mode": "Incremental",
+				"templateLink": {
+					"uri": "[variables('AKSDeploymentTemplateUrl')]",
+					"contentVersion": "1.0.0.0"
+				},
+				"parameters": {
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/azuredeploy.json:360]: format the json, e.g. SHIFT+ALT+F in VS Code

File: aks-kubesphere-linux/nested/AKS.json [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+            "type": "string",
+            "defaultValue": "azure",
+            "metadata": {
+                "description": "用于构建 Kubernetes 网络的网络策略。"
+            }
+        }
+    },
+    "resources": [
+        {
+            "apiVersion": "2020-09-01",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/nested/AKS.json:129]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: aks-kubesphere-linux/nested/clientnode.json [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+        "publicIpAddressSku": {
+            "type": "string"
+        },
+        "virtualMachineName": {
+            "type": "string"
+        },
+        "virtualMachineComputerName": {
+            "type": "string"
+        },
+        // "virtualMachineRG": {
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/nested/clientnode.json:38]: remove comments

File: aks-kubesphere-linux/nested/template/parameters.json [hunk 14]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/nested/template/parameters.json:1]: remove this file

File: aks-kubesphere-linux/parameters.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
+parameters
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/parameters.md:1]: update this file after removing the defaultValues for usernames/passwords

File: aks-kubesphere-linux/metadata.json [hunk 16]
```diff
@@ -0,0 +1,13 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "aks kubesphere linux 1.0",
+    "description": "aks kubesphere linux 1.0",
+    "summary": "Deploy KubeSphere on AKS",
+    "githubUsername": "yaorui",
+    "dateUpdated": "2021-03-05",
+    "type": "QuickStart",
+    "validationType": "Manual",
```
Comment (bmoore-msft) [inline aks-kubesphere-linux/metadata.json:9]: why manual?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
remove the empty createServicePrincipal.sh file...

Also check the pipeline results for other changes that need to be made.

**Comment by ruiyaoOps** [comment]
> remove the empty createServicePrincipal.sh file...
> 
> Also check the pipeline results for other changes that need to be made.

Thank you very much for your reply and guidance.
I will carefully check and modify to meet the standards.

File: aks-kubesphere-linux/metadata.json [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+    "environments": [
+        "AzureChinaCloud"
```
[the remark this answers is not in the document]
Comment (ruiyaoOps) [reply aks-kubesphere-linux/metadata.json:11]: It's supported in AzureCloud.
The AzureChinaCloud parameters are set for users in China. It used at deployKubeSphere.sh for users in China set "az cloud set -n AzureChinaCloud". 
Maybe the description is not very clear, I will modify this.

File: aks-kubesphere-linux/metadata.json [hunk 18]
```diff
@@ -0,0 +1,13 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "aks kubesphere linux 1.0",
+    "description": "aks kubesphere linux 1.0",
```
[the remark this answers is not in the document]
Comment (ruiyaoOps) [reply aks-kubesphere-linux/metadata.json:4]: This is my oversight, I will add a more detailed explanation. THX

File: aks-kubesphere-linux/azuredeploy.parameter.json (hunk 1 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.parameter.json:6]: Oh THX, I will do it.

File: aks-kubesphere-linux/azuredeploy.json (hunk 2 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:117]: Thanks for your reminder, I will modify it.

File: aks-kubesphere-linux/azuredeploy.json (hunk 3 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:190]: This is my fault. The defaultValue of adminUser and password should not be set. This is not standard. I will check all files carefully.

File: aks-kubesphere-linux/azuredeploy.json (hunk 4 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:196]: I'm so sorry, this is my negligence. During the test, the default password was set and displayed in plain text for the convenience. I will modify these.

File: aks-kubesphere-linux/azuredeploy.json (hunk 5 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:197]: I'm so sorry, I will do it.

File: aks-kubesphere-linux/azuredeploy.json (hunk 7 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:239]: Most Chinese users cannot access git, but Azure Cloud in China can be accessed smoothly. We have carefully checked.

File: aks-kubesphere-linux/README.md (hunk 8 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/README.md:1]: I will do this.

File: aks-kubesphere-linux/README.md (hunk 9 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/README.md:16]: Is so cool, but it failed when my test, I will re-verify this soon.

File: aks-kubesphere-linux/azuredeploy.json (hunk 6 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:220]: I will do it.

File: aks-kubesphere-linux/azuredeploy.json (hunk 10 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:269]: I'm so sorry, I will modify this.

File: aks-kubesphere-linux/azuredeploy.json (hunk 11 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/azuredeploy.json:360]: Oh my god, the mixed-use of tabs and spaces caused this problem, I'm so sorry.

File: aks-kubesphere-linux/nested/AKS.json (hunk 12 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/nested/AKS.json:129]: I will recheck soon.

File: aks-kubesphere-linux/nested/clientnode.json (hunk 13 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/nested/clientnode.json:38]: Oh, I'm so sorry.

File: aks-kubesphere-linux/nested/template/parameters.json (hunk 14 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/nested/template/parameters.json:1]: Oh my god, I forgot it.

File: aks-kubesphere-linux/parameters.md (hunk 15 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/parameters.md:1]: yeah, I will change this.

File: aks-kubesphere-linux/metadata.json (hunk 16 above)
Comment (ruiyaoOps) [reply aks-kubesphere-linux/metadata.json:9]: I think the service principals have used in scripts, so use "Manual", I will review https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md and fix it.

File: aks-kubesphere-linux/README.md (hunk 9 above)
Comment (bmoore-msft) [reply aks-kubesphere-linux/README.md:16]: What was the failure?

File: aks-kubesphere-linux/metadata.json (hunk 16 above)
Comment (bmoore-msft) [reply aks-kubesphere-linux/metadata.json:9]: Ok, if the MSI works (it should) then you can do automated validation - if that doesn't work there are some GEN values for the pipeline that may work (depending on the permissions used) - so we should try those options before resorting to manual validation.

**Comment by ruiyaoOps** [comment]
MSI Works, it's so cool, most of the issues have been fixed,.
But I don't understand how to generate a top of the Readme.md, like Azure Public Test Date, I will continue to review the docs.

And AKS.json about the proper sort order of properties, I had checked, but I don't know what's wrong.

**Comment by ruiyaoOps** [comment]
Oh , I forget my azuredeploy.parameters.json.

**Pull request closed without merging**